lib/eventum/auth/APIAuthToken.php

```diff
@@ -0,0 +1,159 @@
+// +----------------------------------------------------------------------+
+// | Eventum - Issue Tracking System                                      |
+// +----------------------------------------------------------------------+
+// | Copyright (c) 2015 Eventum Team.                                     |
+// |                                                                      |
+// | This program is free software; you can redistribute it and/or modify |
+// | it under the terms of the GNU General Public License as published by |
+// | the Free Software Foundation; either version 2 of the License, or    |
+// | (at your option) any later version.                                  |
+// |                                                                      |
+// | This program is distributed in the hope that it will be useful,      |
+// | but WITHOUT ANY WARRANTY; without even the implied warranty of       |
+// | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the        |
+// | GNU General Public License for more details.                         |
+// |                                                                      |
+// | You should have received a copy of the GNU General Public License    |
+// | along with this program; if not, write to:                           |
+// |                                                                      |
+// | Free Software Foundation, Inc.                                       |
+// | 51 Franklin Street, Suite 330                                        |
+// | Boston, MA 02110-1301, USA.                                          |
+// +----------------------------------------------------------------------+
+
+class APIAuthToken
+{
+    private static $default_alg = 'HS256';
+
+    public static function generate($usr_id)
+    {
+        $factory = new RandomLib\Factory();
+        $generator = $factory->getMediumStrengthGenerator();
+        $token = $generator->generateString(32, \RandomLib\Generator::CHAR_ALNUM);
+
+        self::saveToken($usr_id, $token);
+
+        return $token;
+    }
+
+    public static function saveToken($usr_id, $token)
+    {
+        $sql = 'INSERT INTO
+                    {{%api_token}}
+                SET
+                    apt_usr_id = ?,
+                    apt_created = ?,
+                    apt_token = ?';
+        try {
+            $res = DB_Helper::getInstance()->query($sql, array(
+                $usr_id,
+                Date_Helper::getCurrentDateGMT(),
+                $token,
+            ));
+        } catch (DbException $e) {
+            return -1;
+        }
+    }
+
+    public static function isTokenValidForEmail($token, $email)
+    {
+        try {
+            if (self::checkTokenStatus($token) == User::getUserIDByEmail($email, true)) {
+                return true;
+            }
+        } catch (AuthException $e) {
+            return false;
+        }
+    }
+
+    public static function checkTokenStatus($token)
```

Comment:
should include $usr_id in functin call, and fetch all user tokens with status active and compare them php side using not time leaking algorithm. but that method is private right now in AuthPassword method. so it can be done later when this PR is merged.